pkg/workloads/workload.go

```diff
@@ -0,0 +1,307 @@
+}
+
+// CreateWorkload creates a New Relic One workload.
+func (e *Workloads) CreateWorkload(accountID int, workload *CreateInput) (*Workload, error) {
+	resp := workloadCreateResponse{}
+	vars := map[string]interface{}{
+		"accountId": accountID,
+		"workload":  workload,
+	}
+
+	if err := e.client.Query(createWorkloadMutation, vars, &resp); err != nil {
+		return nil, err
+	}
+
+	return &resp.WorkloadCreate, nil
+}
+
+// DeleteWorkload deletes a New Relic One workload.
+func (e *Workloads) DeleteWorkload(accountID int, workload *DeleteInput) (*Workload, error) {
+	resp := workloadDeleteResponse{}
+	vars := map[string]interface{}{
+		"accountId": accountID,
+		"workload":  workload,
+	}
+
+	if err := e.client.Query(deleteWorkloadMutation, vars, &resp); err != nil {
+		return nil, err
+	}
+
+	return &resp.WorkloadDelete, nil
+}
+
+// DuplicateWorkload duplicates a New Relic One workload.
+func (e *Workloads) DuplicateWorkload(accountID int, workload *DuplicateInput) (*Workload, error) {
+	resp := workloadDuplicateResponse{}
+	vars := map[string]interface{}{
+		"accountId": accountID,
+		"workload":  workload,
+	}
+
+	if err := e.client.Query(duplicateWorkloadMutation, vars, &resp); err != nil {
+		return nil, err
+	}
+
+	return &resp.WorkloadDuplicate, nil
+}
+
+// UpdateWorkload updates a New Relic One workload.
+func (e *Workloads) UpdateWorkload(accountID int, workload *UpdateInput) (*Workload, error) {
+	resp := workloadUpdateResponse{}
+	vars := map[string]interface{}{
+		"accountId": accountID,
+		"workload":  workload,
+	}
+
+	if err := e.client.Query(updateWorkloadMutation, vars, &resp); err != nil {
+		return nil, err
+	}
+
+	return &resp.WorkloadUpdate, nil
+}
+
+const (
+	// graphqlWorkloadStructFields is the set of fields that we want returned on workload queries,
+	// and should map back directly to the Workload struct
+	graphqlWorkloadStructFields = `
+			account {
+				id
+				name
+			}
+			createdBy {
+				email
+				gravatar
+				id
+				name
+			}
+			createdAt
+			entities {
+				guid
+			}
+			entitySearchQueries {
+				createdAt
+				createdBy {
+					email
+					gravatar
+					id
+					name
+				}
+				name
+				id
+				query
+				updatedAt
+			}
+			entitySearchQuery
+			guid
+			id
+			permalink
+			name
+			scopeAccounts {
+				accountIds
+			}
+			updatedAt
+`
+
+	getWorkloadQuery = `query($id: Int!, $accountId: Int!) { actor { account(id: $accountId) { workload { collection(id: $id)  {` +
+		graphqlWorkloadStructFields +
+		` } } } } }`
+
+	listWorkloadsQuery = `query($accountId: Int!) { actor { account(id: $accountId) { workload { collections {` +
+		graphqlWorkloadStructFields +
+		` } } } } }`
+
+	createWorkloadMutation = `
+		mutation($accountId: Int!, $workload: WorkloadCreateInput!) {
+			workloadCreate(accountId: $accountId, workload: $workload) {` +
+		graphqlWorkloadStructFields +
+		` } }`
+
+	deleteWorkloadMutation = `
+		mutation($accountId: Int!, $workload: WorkloadDeleteInput!) {
```

Comment:
Where are the graphql types like `WorkloadDeleteInput` defined?  I was writing queries today and didn't understand this.  I'm trying to do something similar for another endpoint.

Reply:
`WorkloadDeleteInput!` above is the GraphQL type, not Golang.  Look up at L156, where the query is called `var` is passed in as a `map[string]interface{}` where the key is the name of the required variable ("accountId" and "workload") for this example.

`workload` is of Go type `*DeleteInput` defined on L81 which matches Go style better (full name workloads.DeleteInput)

Reply:
Thanks for the explanation.  The variable passing I think I understand, and I guess the types available are just known from a query like this one?
```
{
  __schema {
    types {
      name
    }
  }
}
```

Reply:
I was trying to avoid the redundancy of types that share the package name like `workloads.WorkloadDeleteInput`.  Once we start generating this code I wonder if it would be better to just forget about it.

Reply:
@zlesnr Yes, or building the query via https://api.newrelic.com/graphiql which provides context-sensitive help including type definitions.

@ctrombley I agree with the non-redundant version, and I believe one of the linters looks for these as it's considered sub-optimal.